File: Chapter09/DisplayPlatformInfo/DisplayPlatformInfo/DisplayPlatformInfo.Droid/PlatformInfo.cs
using System;
using Android.OS;
using Xamarin.Forms;

[assembly: Dependency(typeof(DisplayPlatformInfo.Droid.PlatformInfo))]

namespace DisplayPlatformInfo.Droid
{
    public class PlatformInfo : IPlatformInfo
    {
        public string GetModel()
        {
            return String.Format("{0} {1}", Build.Manufacturer, 
                                            Build.Model);
        }

        public string GetVersion()
        {
            return Build.VERSION.Release.ToString();
        }
    }
}

File: Chapter09/FS/DisplayPlatformInfo/Droid/PlatformInfo.fs
namespace DisplayPlatformInfo.Droid

open DisplayPlatformInfo
open Android.OS
open Xamarin.Forms

type PlatformInfo () =
    interface IPlatformInfo with
        member this.GetModel () = sprintf "%s %s" Build.Manufacturer Build.Model

        member this.GetVersion () = Build.VERSION.Release.ToString()

module DisplayPlatform =
    [<assembly: Dependency(typeof<PlatformInfo>)>]
    do()
